### DISCUSSION TITLE
API: GET /api/shopping-lists

### DISCUSSION THREAD

**Discussion by xxxserxxx** [mealie-recipes/mealie#956] [Feature Request]
**Before submitting this feature request I have**
_COMPLETE AND FILL THESE OUT_
- [X] Reviewed the documentation and verified that my feature request isn't already a feature.
- [X] Reviewed the existing feature requests and my feature has not been requested.
- [X] Checked the [tasks taged](https://github.com/hay-kot/mealie/issues?q=is%3Aissue+is%3Aopen+label%3Atask+) issues and verified my feature is not covered

**Please Describe The Problem To Be Solved**
There is no way other than trial-and-error for API users to get a list of valid shopping list IDs. Currently, users need to implement the algorithm:

```
for i = 1; i < SOME_ARBITRARY_LARGE_NUMBER; i++
    GET /api/shopping-lists/{i}
    If response body is `null`
        continue
    Else
        add {i} to list of lists
``` 

**(Optional): Suggest A Solution**
Implement `GET` on `/api/shopping-lists` and return a JSON with the list of IDs. In the most simple case, it could be just an array of IDs:
```
[2,3,5]
```
More practical would be a list of names & ids; this would allow lazy-loading of data in programs. For example:
```
{"shopping-lists": [
   {"name": "List #1 name", "id": 2},
   {"name": "List #2 name", "id": 3},
   {"name": "List #3 Name", "id": 5}
   ]
}
```
Either of these would improve the REST-fulness of the API: a GET on the asset root to retrieve a list of asset references is a common function.

**Additional Information**
- [ ] If the feature is accepted I'm willing to submit a PR to provide this feature

I have no experience with Python. I'll attempt to figure out how the REST interface is generated and submit a PR.

**Update** I pulled the project source and started poking through the code; because of the ORM, it doesn't look straightforward to copy & adapt an existing chunk of code, and it looks as if, for me, it'd require a significant education to contribute a patch.

**Comment by michael-genson** [comment]
Mealie has changed drastically since this discussion was opened, the REST API allows for this (and plenty more!).

I'm currently going through some old discussions, so apologies for the necro.
